### ISSUE TITLE
Cannot go back from notebook editor

### ISSUE THREAD

**Issue by rafpaf** [metabase/metabase#50971]
The user cannot go back from the notebook editor since the confirmation modal appears behind the data picker modal.

![Image](https://github.com/user-attachments/assets/233fbcd4-a956-4efe-89e3-70432df43a4b)

@kamilmielnik noted this in July: https://github.com/metabase/metabase/issues/45469#issuecomment-2232650222

### To Reproduce

1. Go to New > Model
2. Click on Notebook editor
3. Press the back button
4. Notice that the page does not respond

### Severity

P2

**Comment by WiNloSt** [comment]
happens to v50.35, and v51.7
